### PR TITLE
Feature/terrain landscape camera

### PR DESCRIPTION
**Pull request by weberjavi**
This PR sets the view (camera location and position) on terrain view.
[Pivotal task](https://www.pivotaltracker.com/story/show/160683743)
[Designs](https://zpl.io/V4lnkrM)

The terrain view gets triggered when clicking on a spot inside a grid element. Grid elements only appear when passed certain `zoom in` threshold.

Note that this feature is not yet merged with terrain mode sidebar update (just test the camera positioning)

### CODE REVIEW

File: src/app/components/v2/map/map.js [hunk 1]
```diff
@@ -52,18 +53,33 @@ class CesiumComponent extends Component {
...
       if (lockNavigation) disablePanning(this.viewer);
       if (
-        coordinates && (prevProps.coordinates !== coordinates || prevProps.coordinatesOptions !== coordinatesOptions)
+        coordinates &&
+          !terrainMode &&
+          (prevProps.coordinates !== coordinates || prevProps.coordinatesOptions !== coordinatesOptions)
       ) {
         this.setCoordinates();
       }
+      if (terrainMode && prevProps.coordinates === coordinates) {
```
Comment (j8seangel) [inline src/app/components/v2/map/map.js]: Shouldn't this be the latLng property? we are not using coordinates for terrainMode.

File: src/app/pages/root/map/map-component.jsx [hunk 2]
```diff
@@ -86,28 +91,47 @@ class MapComponent extends PureComponent {
   };
 
   handleGridClick = (e, object) => {
+    // Allow pickPosition function to perform a 3D calculation
+    this.map.scene.pickTranslucentDepth = true;
```
Comment (j8seangel) [inline src/app/pages/root/map/map-component.jsx]: 🔮

File: src/app/components/v2/map/map.js (hunk 1 above)
Comment (j8seangel) [reply src/app/components/v2/map/map.js]: Not completely, we are coupling the feature of changing coordinates to the lat&Lng one, so why not check if the latLng is different instead?

Comment (weberjavi) [reply src/app/components/v2/map/map.js]: We are going to remove `latLng` from the URL after escaping `terrainMode`?

Comment (weberjavi) [reply src/app/components/v2/map/map.js]: ⚡️ addressed 
Much better this way.
Thank you!!

Comment (weberjavi) [reply src/app/components/v2/map/map.js]: Is kinda ugly, but being an object have to compare its properties.

**Comment by j8seangel** [comment]
🎉

**Pull request merged**